app/apps/server/bridges/livechat.ts

```diff
@@ -222,6 +223,20 @@ export class AppLivechatBridge extends LivechatBridge {
 		return LivechatDepartment.findEnabledWithAgents().map(boundConverter);
 	}
 
+	protected async _fetchLivechatRoomMessages(appId: string, roomId: string): Promise<Array<IMessage>> {
+		this.orch.debugLog(`The App ${ appId } is getting the transcript for livechat room ${ roomId }.`);
+		const messages = Livechat.getRoomMessages({ rid: roomId });
+
+		const result = [];
+
+		for await (const message of messages) {
+			const res = await this.orch.getConverters()?.get('messages').convertMessage(message);
+			result.push(res);
+		}
```

Comment:
I would suggest this be changed into a `Promise.allSettled()` call, allowing the calls to `convertMessage` to be run concurrently